dispatches/case_studies/renewables_case/run_double_loop_battery.py

```diff
@@ -0,0 +1,308 @@
+from pathlib import Path
+from dispatches_sample_data import rts_gmlc
+from dispatches.case_studies.renewables_case.double_loop_utils import read_rts_gmlc_wind_inputs
+from dispatches.case_studies.renewables_case.prescient_options import *
+from load_parameters import duration
+
+
+if __name__ == "__main__":
+    prescient_options = default_prescient_options.copy()
+
+    usage = "Run double loop simulation with RE model."
+    parser = ArgumentParser(usage)
+
+    parser.add_argument(
+        "--sim_id",
+        dest="sim_id",
+        help="Indicate the simulation ID.",
+        action="store",
+        type=int,
+        default=0,
+    )
+
+    parser.add_argument(
+        "--wind_pmax",
+        dest="wind_pmax",
+        help="Set wind capacity in MW.",
+        action="store",
+        type=float,
+        default=50,
+    )
+
+    parser.add_argument(
+        "--battery_energy_capacity",
+        dest="battery_energy_capacity",
+        help="Set the battery energy capacity in MWh.",
+        action="store",
+        type=float,
+        default=20.0,
+    )
+
+    parser.add_argument(
+        "--battery_pmax",
+        dest="battery_pmax",
+        help="Set the battery power capacity in MW.",
+        action="store",
+        type=float,
+        default=5.0,
+    )
+
+    parser.add_argument(
+        "--n_scenario",
+        dest="n_scenario",
+        help="Set the number of price scenarios.",
+        action="store",
+        type=int,
+        default=3,
+    )
+
+    parser.add_argument(
+        "--participation_mode",
+        dest="participation_mode",
+        help="Indicate the market participation mode.",
+        action="store",
+        type=str,
+        default="Bid",
+    )
+
+    options = parser.parse_args()
+
+    sim_id = options.sim_id
+    wind_pmax = options.wind_pmax
+    battery_energy_capacity = options.battery_energy_capacity
+    battery_pmax = options.battery_pmax
+    n_scenario = options.n_scenario
+    participation_mode = options.participation_mode
+    p_min = 0
+
+    allowed_participation_modes = {"Bid", "SelfSchedule"}
+    if participation_mode not in allowed_participation_modes:
+        raise ValueError(
+            f"The provided participation mode {participation_mode} is not supported."
+        )
+
+    wind_df = read_rts_gmlc_wind_inputs(rts_gmlc.source_data_path, wind_generator)
+    wind_df = wind_df[wind_df.index >= start_date]
+    gen_capacity_factor = wind_df[f"{wind_generator}-RTCF"].values.tolist()
+
+    # NOTE: `rts_gmlc_data_dir` should point to a directory containing RTS-GMLC scenarios
+    rts_gmlc_data_dir = rts_gmlc.source_data_path
+    output_dir = Path(f"new_wind_battery_ratio_duration_sweep_sb/battery_duration_{duration}/Sweep_rerun_wind_battery_stochastic_bidder_sim_{sim_id}")
+    # output_dir = Path(sim_name)
+
+    solver = pyo.SolverFactory(solver_name)
+
+    if participation_mode == "Bid":
+        thermal_generator_params = {
```

Comment:
Please add units and/or a brief description in the comments for these options.

Reply:
The description of each of these inputs should go in the `ThermalGeneratorModelData` class in IDAES

Reply:
Ben has kindly written some of them down in the DISPATCHES Box Notes:

a. min_down_time : Once stopped, the number of time periods the generator must remain off (hours)

b. min_up_time : Once started, the number of time periods the generator must remain on (hours)

c. ramp_up_60min : Ramp up capability in an hour (MW/hr)

d. ramp_down_60min : Ramp down capability in an hour (MW/hr)

e. shutdown_capacity : Maximum power available given at start up time (for an hourly time horizon)

f.  startup_capacity : Maximum power available given at shutdown time (for an hourly time horizon)

g. initial_status : Number of time periods the generator has been on (>0) or off (<0) at the initial time period (beginning of the simulation)

h. initial_p_output : Real power output at the time preceding the initial time period (beginning of the simulation)